socketcan_interface/CMakeLists.txt

```diff
@@ -1,151 +1,132 @@
-cmake_minimum_required(VERSION 2.8.3)
+cmake_minimum_required(VERSION 3.5)
 project(socketcan_interface)
 
 if(NOT CMAKE_CXX_STANDARD)
   set(CMAKE_CXX_STANDARD_REQUIRED ON)
   set(CMAKE_CXX_STANDARD 14)
 endif()
 
-find_package(catkin REQUIRED
-  COMPONENTS
-    class_loader
-)
-
-find_package(console_bridge REQUIRED)
-
+find_package(ament_cmake REQUIRED)
+find_package(pluginlib REQUIRED)
 find_package(Boost REQUIRED
   COMPONENTS
     chrono
     system
     thread
 )
-
 find_package(Threads REQUIRED)
 
-catkin_package(
-  INCLUDE_DIRS
-    include
-  LIBRARIES
-    ${PROJECT_NAME}_string
-  CATKIN_DEPENDS
-  DEPENDS
-    Boost
-    console_bridge
-)
-
-include_directories(
-  include
+include_directories(include
+  ${pluginlib_INCLUDE_DIRS}
   ${Boost_INCLUDE_DIRS}
-  ${catkin_INCLUDE_DIRS}
-  ${console_bridge_INCLUDE_DIRS}
 )
 
 # ${PROJECT_NAME}_string
-add_library(${PROJECT_NAME}_string
-    src/string.cpp
+add_library(${PROJECT_NAME}_string SHARED
```

Comment:
does this really have to be shared?

Reply:
Since it is used in builds further down which are then used themselves in further builds, it avoids the linker thinking there are multiple definitions of the functions included in this library. I originally tried doing this without it being SHARED but got multiple linker errors because of this.

Reply:
It *should* work with static as well, but I will have a look later.